d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -219,6 +219,7 @@ public void onChanged(XdsClient.NodeUpdate update)
           }
           catch (PropertySerializationException e)
           {
+            _serviceEventBus.publishInitialize(serviceName, null); // notify the watchers that the service read the data from the cache directly.
```

Comment:
this comment is confusing. Could be "still notify event bus to avoid timeout in case some subscribers are waiting for the data"

Reply:
Thanks. Updated the comment